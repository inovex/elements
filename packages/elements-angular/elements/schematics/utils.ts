import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
