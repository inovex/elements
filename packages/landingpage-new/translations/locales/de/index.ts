import learn from './learn';
import home from './home';
import about from './about';

export default {
  learn,
  home,
  about,
};
