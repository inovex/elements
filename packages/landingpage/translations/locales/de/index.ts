import about from './about.json';
import common from './common.json';
import explore from './explore.json';
import home from './home.json';
import library from './library.json';

export default {
  home,
  about,
  common,
  library,
  explore,
};
