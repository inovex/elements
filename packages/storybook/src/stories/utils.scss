/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
