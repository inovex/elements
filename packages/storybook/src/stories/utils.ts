import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
