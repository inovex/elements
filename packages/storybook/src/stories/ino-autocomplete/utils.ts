import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
