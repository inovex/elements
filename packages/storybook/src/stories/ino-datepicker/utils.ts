import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
