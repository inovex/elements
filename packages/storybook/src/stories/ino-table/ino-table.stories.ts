import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';

  // == event block
const checkboxHandler = e => {
  const el = e.target;
  if (el.tagName.toLowerCase() !== 'ino-checkbox') {
    return;
  }

  const row = el.closest('ino-table-row');
  el.checked = !el.checked;
  row.inoSelected = el.checked;
  if (el.id === 'headerBox') {
    el.indeterminate = false;
    setAllBoxes();
  } else {
    setHeaderBox();
  }
};

const setAllBoxes = () => {
  const mainBox = document.getElementById('headerBox') as HTMLInoCheckboxElement;
  const boxes = Array.from(
    document.querySelectorAll('ino-checkbox:not(#headerBox)')
  ) as HTMLInoCheckboxElement[];
  boxes.forEach(checkbox => {
    checkbox.checked = mainBox.checked;
    checkbox.closest('ino-table-row').selected = mainBox.checked;
  });
};

const setHeaderBox = () => {
  const mainBox = document.getElementById('headerBox') as HTMLInoCheckboxElement;
  const boxes = Array.from(
    document.querySelectorAll('ino-checkbox:not(#headerBox)')
  ) as HTMLInoCheckboxElement[];
  if (boxes.every((i) => i.checked)) {
    mainBox.checked = true;
    mainBox.indeterminate = false;
  } else if (boxes.some((i) => i.checked)) {
    mainBox.indeterminate = true;
  } else {
    mainBox.checked = false;
    mainBox.indeterminate = false;
  }
};

export default {
  title: `Structure/ino-table`,
  component: 'ino-table',
  decorators: [
    defaultDecorator, 
    story => {
      useEffect(() => {
        document.addEventListener('checkedChange', checkboxHandler);
        return () =>  document.removeEventListener('checkedChange', checkboxHandler);;
      });
      return story();
    }
  ],
};

export const Playground: Story<Components.InoTable> = args => html`
  <ino-table>
    <ino-table-row slot="header" header-row>
      <ino-table-cell>ID</ino-table-cell>
      <ino-table-cell>Name</ino-table-cell>
      <ino-table-cell>Release Date</ino-table-cell>
      <ino-table-cell>Box Office</ino-table-cell>
      <ino-table-cell>Rating</ino-table-cell>
    </ino-table-row>

    <ino-table-row>
      <ino-table-cell numeric>1</ino-table-cell>
      <ino-table-cell>The Bourne Identity</ino-table-cell>
      <ino-table-cell>14.06.2002</ino-table-cell>
      <ino-table-cell>$214 million</ino-table-cell>
      <ino-table-cell>93%</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell numeric>2</ino-table-cell>
      <ino-table-cell>Sully</ino-table-cell>
      <ino-table-cell>09.09.2016</ino-table-cell>
      <ino-table-cell>$240.8 million</ino-table-cell>
      <ino-table-cell>84%</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell numeric>3</ino-table-cell>
      <ino-table-cell>The Martian</ino-table-cell>
      <ino-table-cell>02.10.2015</ino-table-cell>
      <ino-table-cell>$630.2 million</ino-table-cell>
      <ino-table-cell>94%</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell numeric>4</ino-table-cell>
      <ino-table-cell>John Wick: Chapter 3</ino-table-cell>
      <ino-table-cell>17.05.2019</ino-table-cell>
      <ino-table-cell>$326.7 million</ino-table-cell>
      <ino-table-cell>92%</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell numeric>5</ino-table-cell>
      <ino-table-cell>10 Cloverfield Lane</ino-table-cell>
      <ino-table-cell>11.03.2016</ino-table-cell>
      <ino-table-cell>$110.2 million</ino-table-cell>
      <ino-table-cell>87%</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell numeric>6</ino-table-cell>
      <ino-table-cell>Thor: Ragnarok</ino-table-cell>
      <ino-table-cell>10.10.2017</ino-table-cell>
      <ino-table-cell>$854 million</ino-table-cell>
      <ino-table-cell>92%</ino-table-cell>
    </ino-table-row>
  </ino-table>
`;

export const WithCheckboxes = () => html`
  <ino-table>
    <ino-table-row slot="header" header-row>
      <ino-table-cell><ino-checkbox id="headerBox" selection></ino-checkbox></ino-table-cell>
      <ino-table-cell>Header A</ino-table-cell>
      <ino-table-cell>Header B</ino-table-cell>
      <ino-table-cell>Header C</ino-table-cell>
      <ino-table-cell>Header D</ino-table-cell>
      <ino-table-cell>Actions</ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell><ino-checkbox selection></ino-checkbox></ino-table-cell>
      <ino-table-cell numeric>1</ino-table-cell>
      <ino-table-cell>Cell B1</ino-table-cell>
      <ino-table-cell>Cell C1</ino-table-cell>
      <ino-table-cell>Cell D1</ino-table-cell>
      <ino-table-cell><ino-icon-button icon="download"></ino-icon-button></ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell><ino-checkbox selection></ino-checkbox></ino-table-cell>
      <ino-table-cell numeric>2</ino-table-cell>
      <ino-table-cell>Cell B2</ino-table-cell>
      <ino-table-cell>Cell C2</ino-table-cell>
      <ino-table-cell>Cell D2</ino-table-cell>
      <ino-table-cell><ino-icon-button icon="download"></ino-icon-button></ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell><ino-checkbox selection></ino-checkbox></ino-table-cell>
      <ino-table-cell numeric>3</ino-table-cell>
      <ino-table-cell>Cell B3</ino-table-cell>
      <ino-table-cell>Cell C3</ino-table-cell>
      <ino-table-cell>Cell D3</ino-table-cell>
      <ino-table-cell><ino-icon-button icon="download"></ino-icon-button></ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell><ino-checkbox selection></ino-checkbox></ino-table-cell>
      <ino-table-cell numeric>4</ino-table-cell>
      <ino-table-cell>Cell B4</ino-table-cell>
      <ino-table-cell>Cell C4</ino-table-cell>
      <ino-table-cell>Cell D4</ino-table-cell>
      <ino-table-cell><ino-icon-button icon="download"></ino-icon-button></ino-table-cell>
    </ino-table-row>
    <ino-table-row>
      <ino-table-cell><ino-checkbox selection></ino-checkbox></ino-table-cell>
      <ino-table-cell numeric>5</ino-table-cell>
      <ino-table-cell>Cell B5</ino-table-cell>
      <ino-table-cell>Cell C5</ino-table-cell>
      <ino-table-cell>Cell D5</ino-table-cell>
      <ino-table-cell><ino-icon-button icon="download"></ino-icon-button></ino-table-cell>
    </ino-table-row>
  </ino-table>
`;