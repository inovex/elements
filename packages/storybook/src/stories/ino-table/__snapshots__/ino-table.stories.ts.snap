// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-table Loading smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table loading="true"
               sort-column-id="id"
               sort-direction="asc"
               no-hover="false"
               sticky-header="false"
    >
      <ino-progress-bar slot="loading-indicator"
                        indeterminate
                        debounce="200"
                        active
      >
      </ino-progress-bar>
      <tr slot="header-row">
      </tr>
      <ino-table-header-cell column-id="id"
                             sort-start="asc"
                             label="ID"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="name"
                             sort-start="asc"
                             label="Name"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="release"
                             label="Release Date"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="box-office"
                             label="Box Office"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="rating"
                             label="Rating"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another1"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another2"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <tr>
        <td>
          1
        </td>
        <td>
          The Bourne Identity
        </td>
        <td>
          14.06.2002
        </td>
        <td>
          $214 million
        </td>
        <td>
          93%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Sully
        </td>
        <td>
          09.09.2016
        </td>
        <td>
          $240.8 million
        </td>
        <td>
          84%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          The Martian
        </td>
        <td>
          02.10.2015
        </td>
        <td>
          $630.2 million
        </td>
        <td>
          94%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          John Wick: Chapter 3
        </td>
        <td>
          17.05.2019
        </td>
        <td>
          $326.7 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          10 Cloverfield Lane
        </td>
        <td>
          11.03.2016
        </td>
        <td>
          $110.2 million
        </td>
        <td>
          87%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Thor: Ragnarok
        </td>
        <td>
          10.10.2017
        </td>
        <td>
          $854 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          The latest movie
        </td>
        <td>
          01.01.2021
        </td>
        <td>
          $900 million
        </td>
        <td>
          95%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table NoHover smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table loading="false"
               sort-column-id="id"
               sort-direction="asc"
               no-hover="true"
               sticky-header="false"
    >
      <tr slot="header-row">
      </tr>
      <ino-table-header-cell column-id="id"
                             sort-start="asc"
                             label="ID"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="name"
                             sort-start="asc"
                             label="Name"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="release"
                             label="Release Date"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="box-office"
                             label="Box Office"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="rating"
                             label="Rating"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another1"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another2"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <tr>
        <td>
          1
        </td>
        <td>
          The Bourne Identity
        </td>
        <td>
          14.06.2002
        </td>
        <td>
          $214 million
        </td>
        <td>
          93%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Sully
        </td>
        <td>
          09.09.2016
        </td>
        <td>
          $240.8 million
        </td>
        <td>
          84%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          The Martian
        </td>
        <td>
          02.10.2015
        </td>
        <td>
          $630.2 million
        </td>
        <td>
          94%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          John Wick: Chapter 3
        </td>
        <td>
          17.05.2019
        </td>
        <td>
          $326.7 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          10 Cloverfield Lane
        </td>
        <td>
          11.03.2016
        </td>
        <td>
          $110.2 million
        </td>
        <td>
          87%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Thor: Ragnarok
        </td>
        <td>
          10.10.2017
        </td>
        <td>
          $854 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          The latest movie
        </td>
        <td>
          01.01.2021
        </td>
        <td>
          $900 million
        </td>
        <td>
          95%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table loading="false"
               sort-column-id="id"
               sort-direction="asc"
               no-hover="false"
               sticky-header="false"
    >
      <tr slot="header-row">
      </tr>
      <ino-table-header-cell column-id="id"
                             sort-start="asc"
                             label="ID"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="name"
                             sort-start="asc"
                             label="Name"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="release"
                             label="Release Date"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="box-office"
                             label="Box Office"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="rating"
                             label="Rating"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another1"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another2"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <tr>
        <td>
          1
        </td>
        <td>
          The Bourne Identity
        </td>
        <td>
          14.06.2002
        </td>
        <td>
          $214 million
        </td>
        <td>
          93%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Sully
        </td>
        <td>
          09.09.2016
        </td>
        <td>
          $240.8 million
        </td>
        <td>
          84%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          The Martian
        </td>
        <td>
          02.10.2015
        </td>
        <td>
          $630.2 million
        </td>
        <td>
          94%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          John Wick: Chapter 3
        </td>
        <td>
          17.05.2019
        </td>
        <td>
          $326.7 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          10 Cloverfield Lane
        </td>
        <td>
          11.03.2016
        </td>
        <td>
          $110.2 million
        </td>
        <td>
          87%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Thor: Ragnarok
        </td>
        <td>
          10.10.2017
        </td>
        <td>
          $854 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          The latest movie
        </td>
        <td>
          01.01.2021
        </td>
        <td>
          $900 million
        </td>
        <td>
          95%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table SelectionWithCheckboxes smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table class="selectable-table">
      <tr slot="header-row">
        <td class="ino-table__cell--checkbox">
          <ino-checkbox id="headerBox"
                        selection
          >
          </ino-checkbox>
        </td>
        <td>
          Header A
        </td>
        <td>
          Header B
        </td>
        <td>
          Header C
        </td>
        <td>
          Header D
        </td>
        <td>
          Header E
        </td>
        <td>
          Header F
        </td>
        <td>
          Header G
        </td>
        <td>
          Header H
        </td>
        <td>
          Header I
        </td>
      </tr>
      <tr>
        <td class="ino-table__cell--checkbox">
          <ino-checkbox selection>
          </ino-checkbox>
        </td>
        <td class="ino-table__cell--numeric">
          1
        </td>
        <td>
          Cell B1
        </td>
        <td>
          Cell C1
        </td>
        <td>
          Cell D1
        </td>
        <td>
          Cell E1
        </td>
        <td>
          Cell F1
        </td>
        <td>
          Cell G1
        </td>
        <td>
          Cell H1
        </td>
        <td>
          Cell I1
        </td>
      </tr>
      <tr class="ino-table__row--selected">
        <td class="ino-table__cell--checkbox">
          <ino-checkbox selection
                        checked
          >
          </ino-checkbox>
        </td>
        <td class="ino-table__cell--numeric">
          2
        </td>
        <td>
          Cell B2
        </td>
        <td>
          Cell C2
        </td>
        <td>
          Cell D2
        </td>
        <td>
          Cell E2
        </td>
        <td>
          Cell F2
        </td>
        <td>
          Cell G2
        </td>
        <td>
          Cell H2
        </td>
        <td>
          Cell I2
        </td>
      </tr>
      <tr>
        <td class="ino-table__cell--checkbox">
          <ino-checkbox selection>
          </ino-checkbox>
        </td>
        <td class="ino-table__cell--numeric">
          3
        </td>
        <td>
          Cell B3
        </td>
        <td>
          Cell C3
        </td>
        <td>
          Cell D3
        </td>
        <td>
          Cell E3
        </td>
        <td>
          Cell F3
        </td>
        <td>
          Cell G3
        </td>
        <td>
          Cell H3
        </td>
        <td>
          Cell I3
        </td>
      </tr>
      <tr class="ino-table__row--active">
        <td class="ino-table__cell--checkbox">
          <ino-checkbox selection>
          </ino-checkbox>
        </td>
        <td class="ino-table__cell--numeric">
          4
        </td>
        <td>
          Cell B4
        </td>
        <td>
          Cell C4
        </td>
        <td>
          Cell D4
        </td>
        <td>
          Cell E4
        </td>
        <td>
          Cell F4
        </td>
        <td>
          Cell G4
        </td>
        <td>
          Cell H4
        </td>
        <td>
          Cell I4
        </td>
      </tr>
      <tr>
        <td class="ino-table__cell--checkbox">
          <ino-checkbox selection>
          </ino-checkbox>
        </td>
        <td class="ino-table__cell--numeric">
          5
        </td>
        <td>
          Cell B5
        </td>
        <td>
          Cell C5
        </td>
        <td>
          Cell D5
        </td>
        <td>
          Cell E5
        </td>
        <td>
          Cell F5
        </td>
        <td>
          Cell G5
        </td>
        <td>
          Cell H5
        </td>
        <td>
          Cell I5
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table SortColumnIdAndSortDirection smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table loading="false"
               sort-column-id="id"
               sort-direction="asc"
               no-hover="false"
               sticky-header="false"
    >
      <tr slot="header-row">
      </tr>
      <ino-table-header-cell column-id="id"
                             sort-start="asc"
                             label="ID"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="name"
                             sort-start="asc"
                             label="Name"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="release"
                             label="Release Date"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="box-office"
                             label="Box Office"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="rating"
                             label="Rating"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another1"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another2"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <tr>
        <td>
          1
        </td>
        <td>
          The Bourne Identity
        </td>
        <td>
          14.06.2002
        </td>
        <td>
          $214 million
        </td>
        <td>
          93%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Sully
        </td>
        <td>
          09.09.2016
        </td>
        <td>
          $240.8 million
        </td>
        <td>
          84%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          The Martian
        </td>
        <td>
          02.10.2015
        </td>
        <td>
          $630.2 million
        </td>
        <td>
          94%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          John Wick: Chapter 3
        </td>
        <td>
          17.05.2019
        </td>
        <td>
          $326.7 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          10 Cloverfield Lane
        </td>
        <td>
          11.03.2016
        </td>
        <td>
          $110.2 million
        </td>
        <td>
          87%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Thor: Ragnarok
        </td>
        <td>
          10.10.2017
        </td>
        <td>
          $854 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          The latest movie
        </td>
        <td>
          01.01.2021
        </td>
        <td>
          $900 million
        </td>
        <td>
          95%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table StickyHeader smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table">
    <ino-table loading="false"
               sort-column-id="id"
               sort-direction="asc"
               no-hover="false"
               sticky-header="true"
    >
      <tr slot="header-row">
      </tr>
      <ino-table-header-cell column-id="id"
                             sort-start="asc"
                             label="ID"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="name"
                             sort-start="asc"
                             label="Name"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="release"
                             label="Release Date"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="box-office"
                             label="Box Office"
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="rating"
                             label="Rating"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another1"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <ino-table-header-cell column-id="another2"
                             label="Another Column"
                             not-sortable
      >
      </ino-table-header-cell>
      <tr>
        <td>
          1
        </td>
        <td>
          The Bourne Identity
        </td>
        <td>
          14.06.2002
        </td>
        <td>
          $214 million
        </td>
        <td>
          93%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Sully
        </td>
        <td>
          09.09.2016
        </td>
        <td>
          $240.8 million
        </td>
        <td>
          84%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          The Martian
        </td>
        <td>
          02.10.2015
        </td>
        <td>
          $630.2 million
        </td>
        <td>
          94%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          John Wick: Chapter 3
        </td>
        <td>
          17.05.2019
        </td>
        <td>
          $326.7 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          10 Cloverfield Lane
        </td>
        <td>
          11.03.2016
        </td>
        <td>
          $110.2 million
        </td>
        <td>
          87%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Thor: Ragnarok
        </td>
        <td>
          10.10.2017
        </td>
        <td>
          $854 million
        </td>
        <td>
          92%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          The latest movie
        </td>
        <td>
          01.01.2021
        </td>
        <td>
          $900 million
        </td>
        <td>
          95%
        </td>
        <td>
          Some information
        </td>
        <td>
          Some information
        </td>
      </tr>
    </ino-table>
  </div>
</div>
`;
